azuki-core/src/main/kotlin/com/anaplan/engineering/azuki/core/runner/JUnitScenarioRunner.kt

```diff
@@ -293,16 +315,28 @@ class JUnitScenarioRunner<
                 ScenarioRun(method.name, method, implementationInstance)
             }
         }
-        return (eacs + adapterTests + analysisScenarios + generatedScenarios + modellingExamples).toMutableList()
+        val nonParameterizedRuns = eacs + adapterTests + analysisScenarios + generatedScenarios + modellingExamples
+        return (if (parameterMethod == null) nonParameterizedRuns else parameterize(nonParameterizedRuns)).toMutableList()
     }
 
-    override fun describeChild(child: ScenarioRun<AF, CF, QF, AGF>): Description =
-        if (excludeImplFromDescription) {
-            Description.createTestDescription(testClass.name, child.method.name)
+    private fun parameterize(baseRuns: List<ScenarioRun<AF, CF, QF, AGF>>): List<ScenarioRun<AF, CF, QF, AGF>> {
+        val parameterPermutations =
+            parameterMethod!!.method.invoke(kClass.companionObjectInstance!!) as? Collection<Array<Any>>
+                ?: throw IllegalStateException("Parameter method $parameterMethod. returns object with invalid type")
+        Log.debug("Test is parameterized, parameter method: ${parameterMethod?.name}, permutation count: ${parameterPermutations.size}")
+        return baseRuns.flatMap { baseRun -> parameterPermutations.map { perm -> baseRun.copy(parameters = perm) } }
```

Comment:
Indeed, that seems to be the case.

Reply:
I did think about doing it like testng, but it seems more sensible to stay consistent with junit